tests/zenko_tests/node_tests/backbeat/ReplicationUtility.js

```diff
@@ -83,20 +83,39 @@ class ReplicationUtility {
     }
 
     deleteAllBlobs(containerName, keyPrefix, cb) {
-        const options = { include: 'metadata' };
-        this.azure.listBlobsSegmented(
-            containerName,
-            null,
-            options,
-            (err, result) => {
-                if (err) {
-                    return cb(err);
-                }
-                // Only delete the blobs put by the current test.
-                const filteredEntries = result.entries.filter(entry => entry.name.startsWith(keyPrefix));
-                return this._deleteBlobList(filteredEntries, containerName, cb);
-            },
-        );
+        const options = { includeMetadata: true };
+        // const iter = this.azure.getContainerClient(containerName)
+        //     .listBlobsFlat(options)
+        //     .byPage(/* { maxPageSize: 200 } */);
+        // const processPage = res => {
+        //     if (res.done) {
+        //         return cb(null);
+        //     }
+        //     const { blobItems } = res.value.segment;
+        //     const filteredEntries = blobItems.filter(blobItem => blobItem.name.startsWith(keyPrefix));
+        //     return this._deleteBlobList(filteredEntries, containerName, (err) => {
+        //         if (err) {
+        //             return cb(err);
+        //         }
+        //         return iter.next().then(processPage, cb);
+        //     });
+        // };
+        // iter.next().then(processPage, cb);
+
+        (async () => {
+            const iter = this.azure.getContainerClient(containerName)
+                .listBlobsFlat(options)
+                .byPage({ maxPageSize: 20 });
+
+            // eslint-disable-next-line no-restricted-syntax
+            for await (const response of iter) {
+                const { blobItems } = response.segment;
+                const filteredEntries = blobItems.filter(blobItem => blobItem.name.startsWith(keyPrefix));
+                await Promise.all(filteredEntries.map(
+                    blob => this.azure.getContainerClient(containerName).deleteBlob(blob, options),
+                ));
+            }
+        })().then(() => cb(null), cb);
```

Comment:
No error handling, is it on purpose?

Reply:
error handling is on caller side: on success, we call the callback with 'null' error ; and on failure we forward the error to the callback.